kolibri/plugins/user_profile/assets/src/views/ChangeFacility/SelectFacility.vue

```diff
@@ -0,0 +1,279 @@
+
+
+<script>
+
+  import { useLocalStorage } from '@vueuse/core';
+  import { computed } from 'kolibri.lib.vueCompositionApi';
+  import responsiveWindowMixin from 'kolibri.coreVue.mixins.responsiveWindowMixin';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import commonSyncElements from 'kolibri.coreVue.mixins.commonSyncElements';
+  import client from 'kolibri.client';
+  import urls from 'kolibri.urls';
+  import BottomAppBar from 'kolibri.coreVue.components.BottomAppBar';
+  import useSavedAddresses from '../../../../../../core/assets/src/views/sync/SelectAddressModalGroup/useSavedAddresses.js';
+  import useDynamicAddresses from '../../../../../../core/assets/src/views/sync/SelectAddressModalGroup/useDynamicAddresses.js';
+  import AddAddressForm from '../../../../../../core/assets/src/views/sync/SelectAddressModalGroup/AddAddressForm';
+
+  export default {
+    name: 'SelectFacility',
+    metaInfo() {
+      return {
+        title: this.$tr('documentTitle'),
+      };
+    },
+    components: { AddAddressForm, BottomAppBar },
+
+    mixins: [responsiveWindowMixin, commonCoreStrings, commonSyncElements],
+    setup(props, context) {
+      const {
+        addresses: discoveredAddresses,
+        discoveringPeers,
+        discoveryFailed,
+        discoveredAddressesInitiallyFetched,
+      } = useDynamicAddresses(props);
+      const {
+        addresses: savedAddresses,
+        removeSavedAddress,
+        refreshSavedAddressList,
+        savedAddressesInitiallyFetched,
+        requestsFailed,
+        deletingAddress,
+        fetchingAddresses,
+      } = useSavedAddresses(props, context);
+      const combinedAddresses = computed(() => {
+        return [...savedAddresses.value, ...discoveredAddresses.value];
+      });
+      const initialFetchingComplete = computed(() => {
+        return savedAddressesInitiallyFetched.value && discoveredAddressesInitiallyFetched.value;
+      });
+      const storageFacilityId = useLocalStorage('kolibri-lastSelectedFacilityId', '');
+      return {
+        combinedAddresses,
+        initialFetchingComplete,
+        discoveredAddresses,
+        discoveryFailed,
+        discoveringPeers,
+        discoveredAddressesInitiallyFetched,
+        savedAddresses,
+        savedAddressesInitiallyFetched,
+        removeSavedAddress,
+        refreshSavedAddressList,
+        requestsFailed,
+        deletingAddress,
+        fetchingAddresses,
+        storageFacilityId,
+      };
+    },
+    data() {
+      return {
+        availableAddressIds: [],
+        availableFacilities: [],
+        selectedFacilityId: '',
+        showAddAddressModal: false,
+      };
+    },
+    inject: ['changeFacilityService'],
+    computed: {
+      isAddressAvailable() {
+        return function(addressId) {
+          return Boolean(this.availableAddressIds.find(id => id === addressId));
+        };
+      },
+    },
+    watch: {
+      selectedFacilityId(newVal) {
```

Comment:
I'd suggest comparing the previous value with the new value and only executing the code below if those two values are different to prevent unnecessary actions.

```js
selectedFacilityId(newVal, oldVal) {
  if (newValue === oldVal) { return; }
  ...
}

Reply:
are you sure this is needed if `newVal` is just a string and not an object? I think the watcher is not triggered unless `selectedFacilityId` changes

Reply:
No, I'm not 100% certain about the object vs string behavior, would need to try. If you're sure it's not triggered, then no problem.